operator/pkg/component/controlplane/control_plane.go

```diff
@@ -89,20 +94,30 @@ func NewIstioOperator(installSpec *v1alpha1.IstioOperatorSpec, translator *trans
 		o.Namespace = defaultIfEmpty(c.Namespace, iop.Namespace(installSpec))
 		out.components = append(out.components, component.NewEgressComponent(c.Name, idx, &o))
 	}
-	for _, cn := range orderedKeys(installSpec.AddonComponents) {
+	addonNames, err := helm.GetAddonNamesFromCharts(installSpec.GetInstallPackagePath())
```

Comment:
i suggest keeping Addons as the Component here and appending the individual component manifests to the slice. Treating the addons as individual components is not really possible because they don't implement IstioComponent interface.

Reply:
They do implement the interface, that's part of what's been implemented here